From 3adc8bf2d7b65eb159ac122f827af369dc741af0 Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Tue, 2 Jun 2020 23:05:13 -0500
Subject: [PATCH 1/2] Return WIFI_ERROR_NOT_SUPPORTED directly in ring buffer
 debugging API

FMAC does not support ring buffer debugging, so just returns
WIFI_ERROR_NOT_SUPPORTED in ring buffer debugging related functions.

Signed-off-by: Wright Feng <wright.feng@cypress.com>
---
 bcmdhd/wifi_hal/wifi_logger.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/bcmdhd/wifi_hal/wifi_logger.cpp b/bcmdhd/wifi_hal/wifi_logger.cpp
index 7a38883..2eae834 100644
--- a/bcmdhd/wifi_hal/wifi_logger.cpp
+++ b/bcmdhd/wifi_hal/wifi_logger.cpp
@@ -432,17 +432,24 @@ wifi_error wifi_get_driver_version(wifi_interface_handle iface, char *buffer, in
 /* API to collect driver records */
 wifi_error wifi_get_ring_data(wifi_interface_handle iface, char *ring_name)
 {
+    /* FMAC does not support ring buffer debugging */
+    return WIFI_ERROR_NOT_SUPPORTED;
+    /*
     DebugCommand *cmd = new DebugCommand(iface, ring_name, GET_RING_DATA);
     NULL_CHECK_RETURN(cmd, "memory allocation failure", WIFI_ERROR_OUT_OF_MEMORY);
     wifi_error result = (wifi_error)cmd->start();
     cmd->releaseRef();
     return result;
+    */
 }
 
 /* API to get the status of all ring buffers supported by driver */
 wifi_error wifi_get_ring_buffers_status(wifi_interface_handle iface,
         u32 *num_rings, wifi_ring_buffer_status *status)
 {
+    /* FMAC does not support ring buffer debugging */
+    return WIFI_ERROR_NOT_SUPPORTED;
+    /*
     if (status && num_rings) {
         DebugCommand *cmd = new DebugCommand(iface, num_rings, status, GET_RING_STATUS);
         NULL_CHECK_RETURN(cmd, "memory allocation failure", WIFI_ERROR_OUT_OF_MEMORY);
@@ -453,6 +460,7 @@ wifi_error wifi_get_ring_buffers_status(wifi_interface_handle iface,
         ALOGE("Ring status buffer NULL");
         return  WIFI_ERROR_INVALID_ARGS;
     }
+    */
 }
 
 /* API to get supportable feature */
@@ -474,6 +482,9 @@ wifi_error wifi_get_logger_supported_feature_set(wifi_interface_handle iface,
 wifi_error wifi_start_logging(wifi_interface_handle iface, u32 verbose_level,
         u32 flags, u32 max_interval_sec, u32 min_data_size, char *ring_name)
 {
+    /* FMAC does not support ring buffer debugging */
+    return WIFI_ERROR_NOT_SUPPORTED;
+    /*
     if (ring_name) {
         DebugCommand *cmd = new DebugCommand(iface, verbose_level, flags, max_interval_sec,
                     min_data_size, ring_name, START_RING_LOG);
@@ -485,6 +496,7 @@ wifi_error wifi_start_logging(wifi_interface_handle iface, u32 verbose_level,
         ALOGE("Ring name NULL");
         return  WIFI_ERROR_INVALID_ARGS;
     }
+    */
 }
 
 
-- 
2.25.0

